lxd/storage/drivers/driver_powerflex_utils.go

```diff
@@ -0,0 +1,1021 @@
+		}
+	}
+
+	return nil, api.StatusErrorf(http.StatusNotFound, "Host not found using nqn: %q", nqn)
+}
+
+// createHost creates a new host.
+func (p *powerFlexClient) createHost(name string, nqn string) (string, error) {
+	body, err := p.createBodyReader(map[string]any{
+		"name": name,
+		"nqn":  nqn,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var actualResponse struct {
+		ID string `json:"id"`
+	}
+
+	err = p.requestAuthenticated(http.MethodPost, "/api/types/Host/instances", body, &actualResponse)
+	if err != nil {
+		return "", fmt.Errorf("Failed to create host: %w", err)
+	}
+
+	return actualResponse.ID, nil
+}
+
+// deleteHost deletes the host behind hostID.
+func (p *powerFlexClient) deleteHost(hostID string) error {
+	err := p.requestAuthenticated(http.MethodPost, fmt.Sprintf("/api/instances/Sdc::%s/action/removeSdc", hostID), nil, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to delete host: %w", err)
+	}
+
+	return nil
+}
+
+// createHostVolumeMapping creates the mapping between a host and volume.
+func (p *powerFlexClient) createHostVolumeMapping(hostID string, volumeID string) error {
+	body, err := p.createBodyReader(map[string]any{
+		"hostId": hostID,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = p.requestAuthenticated(http.MethodPost, fmt.Sprintf("/api/instances/Volume::%s/action/addMappedHost", volumeID), body, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to create host volume mapping between %q and %q: %w", hostID, volumeID, err)
+	}
+
+	return nil
+}
+
+// deleteHostVolumeMapping deletes the mapping between a host and volume.
+func (p *powerFlexClient) deleteHostVolumeMapping(hostID string, volumeID string) error {
+	body, err := p.createBodyReader(map[string]any{
+		"hostId": hostID,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = p.requestAuthenticated(http.MethodPost, fmt.Sprintf("/api/instances/Volume::%s/action/removeMappedHost", volumeID), body, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to delete host volume mapping between %q and %q: %w", hostID, volumeID, err)
+	}
+
+	return nil
+}
+
+// getHostVolumeMappings returns the volume mappings for the host behind hostID.
+func (p *powerFlexClient) getHostVolumeMappings(hostID string) ([]powerFlexVolume, error) {
+	var actualResponse []powerFlexVolume
+	err := p.requestAuthenticated(http.MethodGet, fmt.Sprintf("/api/instances/Sdc::%s/relationships/Volume", hostID), nil, &actualResponse)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get host volume mappings: %w", err)
+	}
+
+	return actualResponse, nil
+}
+
+// canNVMe returns true if this node supports the PowerFlex NVMe mode.
+func (d *powerflex) canNVMe() bool {
+	err := util.LoadModule("nvme_fabrics")
+	if err != nil {
+		return false
+	}
+
+	err = util.LoadModule("nvme_tcp")
+	return err == nil
+}
+
+// canSDC returns true if this node supports the PowerFlex SDC mode.
+func (d *powerflex) canSDC() bool {
+	if !shared.PathExists("/bin/emc/scaleio/scini.ko") {
```

Comment:
Although we're going to remove SDC stuff for this PR, for future reference we'll need to use `HostPath()` here as we wont be bundling SDC in the snap.

Reply:
When I first implemented the retrieval of `/etc/nvme/hostnqn` in earlier versions of the driver I could successfully make use of `HostPath()` to load the file. For the checking of the SDC bits I suspect we are good to go then. 

Loading the module might get tricky. Since it is not located in the std path we have to `insmod {path}` it from `/bin/emc/scaleio/scini.ko`. But we could potentially offload this to the user since there is a systemd service bundled with the SDC that also needs to be present on the system. So maybe this could be a requirement that the module is already loaded.

But for this PR it will get removed.